functions/src/weekly-markets-emails.ts

```diff
@@ -0,0 +1,69 @@
+import * as functions from 'firebase-functions'
+import * as admin from 'firebase-admin'
+
+import { Contract } from '../../common/contract'
+import { getPrivateUser, getValues, isProd, log } from './utils'
+import { filterDefined } from '../../common/util/array'
+import { sendThreeContractsEmail } from './emails'
+import { createRNG, shuffle } from '../../common/util/random'
+import { DAY_MS } from '../../common/util/time'
+
+export const weeklyMarketsEmails = functions
+  .runWith({ secrets: ['MAILGUN_KEY'] })
+  .pubsub.schedule('every 1 minutes')
+  .onRun(async () => {
+    await sendTrendingMarketsEmailsToAllUsers()
+  })
+
+const firestore = admin.firestore()
+
+async function getTrendingContracts() {
+  return await getValues<Contract>(
+    firestore
+      .collection('contracts')
+      .where('isResolved', '==', false)
+      .where('closeTime', '>', Date.now() + DAY_MS)
+      .where('visibility', '==', 'public')
```

Comment:
Do we even use the visibility flag?

Reply:
No time like the present! It is set on markets currently and we've been talking about using it.